core/model/admin.py

```diff
@@ -219,6 +226,43 @@ def remove_role(self, role, library=None):
         if role:
             _db.delete(role)
 
+    def generate_reset_password_token(self, secret_key: str) -> str:
+        serializer = URLSafeTimedSerializer(secret_key)
+
+        return serializer.dumps(self.email, salt=self.password_hashed)
```

Comment:
Using the password hash as the salt... hmmm. Adds an extra layer of one-time use protection, once the password has changed none of the previously generated links will work. Not sure if it's good or bad, but it sure is neat.

Reply:
Haha yeah, I'm also not completely sure if it's good or bad 😂 
Yeah, I wanted to skip saving things to DB but also wanted to invalidate tokens when the password was changed. I was looking at what Django does, they encode the password hash in the token so they use that as a check. However, I wasn't comfortable putting the hash in the token since you can decode itsdangerous tokens (or jwts) without any secrets (even if it is password hash, not real password). And I didn't want to go too deep on security and copy what Django is doing since I'm not sure it is good idea to implement that by myself.